### ISSUE TITLE
Not able to create bottom annotation with gender information

### ISSUE THREAD

**Issue by vincentye2** [jokergoo/ComplexHeatmap#1007]
Please provide example datasets and the code you use. It will help me to understand your problem and help you!

Hi there,

I'm generating Oncoprints from mouse sequencing data. 

As an example this is my code for the data:

`oncoPrint(mat, alter_fun = list(
  background = function(x, y, w, h) 
    grid.rect(x, y, w*0.9, h*0.9, gp = gpar(fill = "#CCCCCC", col = NA)),
  loss = function(x, y, w, h) 
    grid.rect(x, y, w*0.9, h*0.9, gp = gpar(fill = "red", col = NA)),
  gain = function(x, y, w, h) 
    grid.rect(x, y, w*0.9, h*0.9, gp = gpar(fill = "green", col = NA)),
  missense = function(x, y, w, h) 
    grid.points(x, y, pch = 20),
  nonsense = function(x, y, w, h) {
    grid.segments(x - w*0.4, y - h*0.4, x + w*0.4, y + h*0.4, gp = gpar(lwd = 2, col = "blue"))
    grid.segments(x + w*0.4, y - h*0.4, x - w*0.4, y + h*0.4, gp = gpar(lwd = 2, col = "blue"))
  }
), col = c(gain = "green", loss = "red", missense = "black", nonsense = "blue"), remove_empty_columns = FALSE, remove_empty_rows = FALSE, show_column_names = FALSE, column_title = "Endpoint", right_annotation = NULL, row_order = 1:nrow(mat), 
show_row_names = FALSE, heatmap_legend_param = heatmap_legend_param, show_pct = FALSE, bottom_annotation = haEnd)`

Similarly I imported a tab-delimited file with gender information:

anno_dfEnd = read.table("data.txt", header = TRUE, sep = '\t')
anno_dfEnd = as.data.frame(anno_dfEnd)
sexEnd = anno_dfEnd[,"Sex"]
age = as.numeric(anno_dfEnd$Age)
haEnd = HeatmapAnnotation(Sex = sexEnd, col = list(Sex = c("M" = "blue", "F" = "red")))

When I attempt to plot with the bottom annotation I get this error: 

Error: Sex: cannot map colors to some of the levels: M

I would appreciate any help! I have attached my TSV file for reference.
[data.txt](https://github.com/jokergoo/ComplexHeatmap/files/9932886/data.txt)
